src/dom_components/index.ts

```diff
@@ -417,23 +426,35 @@ export default class ComponentManager extends Module {
    * @param {Object} methods Component methods
    * @return {this}
    */
-  addType(type, methods) {
+  addType(type: string, methods: any) {
     const { em } = this;
-    const { model = {}, view = {}, isComponent, extend, extendView, extendFn = [], extendFnView = [] } = methods;
+    const {
+      model = {},
+      view = {},
+      isComponent,
+      extend,
+      extendView,
+      extendFn = [],
+      extendFnView = [],
+    } = methods;
     const compType = this.getType(type);
     const extendType = this.getType(extend);
     const extendViewType = this.getType(extendView);
-    const typeToExtend = extendType ? extendType : compType ? compType : this.getType('default');
+    const typeToExtend = extendType
+      ? extendType
+      : compType
+      ? compType
+      : this.getType('default');
     const modelToExt = typeToExtend.model;
     const viewToExt = extendViewType ? extendViewType.view : typeToExtend.view;
 
     // Function for extending source object methods
-    const getExtendedObj = (fns, target, srcToExt) =>
+    const getExtendedObj = (fns: any[], target: any, srcToExt: any) =>
       fns.reduce((res, next) => {
         const fn = target[next];
         const parentFn = srcToExt.prototype[next];
         if (fn && parentFn) {
-          res[next] = function (...args) {
+          res[next] = (...args: any[]) => {
```

Comment:
Be careful with arrow functions as `this` inside the function changes its meaning (I'll fix it)